src/v2/repositories/implementations/DappStakingRepository.ts

```diff
@@ -252,6 +252,10 @@ export class DappStakingRepository implements IDappStakingRepository {
     if (!DappStakingRepository.isEraSubscribed) {
       DappStakingRepository.isEraSubscribed = true;
       const api = await this.api.getApi();
+      if (!api.query.hasOwnProperty('dappsStaking')) {
```

Comment:
Let's not use `hasOwnProperty` method from now on ([ref](https://ourcodeworld.com/articles/read/1425/how-to-fix-eslint-error-do-not-access-objectprototype-method-hasownproperty-from-target-object-no-prototype-builtins)). You can use `Object.prototype.hasOwnProperty.call` or `hasProperty` function from Astar-js ([code](https://github.com/AstarNetwork/astar.js/blob/8484da66fe096acd8e2aa0e0ab3b9c4feb0d018c/packages/sdk-core/src/modules/common/index.ts#L10)) 

btw, I think you can just change this line to below code should be enough.
``` 
 if (checkIsDappStakingV3(api)) {
...
}
```

Reply:
Fixed in https://github.com/AstarNetwork/astar-apps/pull/1137/commits/17d3fe839290c98b4fc0b669318d499f83a0d8ec